iwyu_output.cc

```diff
@@ -2298,4 +2306,19 @@ size_t IwyuFileInfo::CalculateAndReportIwyuViolations() {
   return num_edits;
 }
 
+bool IsDirectlyIncluded(const NamedDecl* decl, const IwyuFileInfo& includer) {
+  const vector<string> mapped_headers = internal::GetMappedPublicHeaders(
+      internal::GetQualifiedNameAsString(decl),
+      GetFilePath(includer.file_entry()), GetFilePath(decl));
+  if (!mapped_headers.empty()) {
+    auto pred = [&direct_includes =
+                     includer.direct_includes()](const string& header) {
+      return direct_includes.count(header);
+    };
+    return find_if(mapped_headers.begin(), mapped_headers.end(), pred) !=
+           mapped_headers.end();
+  }
+  return includer.direct_includes_as_fileentries().count(GetFileEntry(decl));
```

Comment:
This line speaks for itself, but I think a comment is needed for the case above where mappings exist to flesh out why mappings affect the decision making.